dashboard/app/controllers/api/v1/user_school_infos_controller.rb

```diff
@@ -0,0 +1,58 @@
+class Api::V1::UserSchoolInfosController < ApplicationController
+  before_action :authenticate_user!
+  before_action :load_user_school_info
+  load_and_authorize_resource
+
+  # PATCH /api/v1/users_school_infos/<id>/update_last_confirmation_date
+  def update_last_confirmation_date
+    @user_school_info.update!(last_confirmation_date: DateTime.now)
+
+    head :no_content
+  end
+
+  # PATCH /api/v1/users_school_infos/<id>/update_end_date
+  def update_end_date
+    ActiveRecord::Base.transaction do
+      @user_school_info.update!(end_date: DateTime.now, last_confirmation_date: DateTime.now)
+      @user_school_info.user.update!(properties: {last_seen_school_info_interstitial: DateTime.now})
+
+      head :no_content
+    end
+  end
+
+  # PATCH /api/v1/user_school_infos/<id>/update_school_info_id
+  # A new row is added to the user school infos table when a teacher updates current school to a different school,
+  # then the school_info_id is updated in the users table.
+
+  def update_school_info_id
+    ActiveRecord::Base.transaction do
+      school = @user_school_info.school_info.school
+      unless school
+        school = School.create!(new_school_params)
+        school.school_info.create!(school_info_params)
+      end
+
+      school_info = school.school_info.order(created_at: :desc).first
```

Comment:
Maybe I'm not following... why would there be multiple `school_info` rows for a given School? And why do we always default to the first one?

If the teacher provided a different school info row, don't we want to keep them linked to that same row?

Reply:
Each time a teacher updates the school info, a new row is added to the user school infos table.  The is how we will track all the schools a teacher has taught at.

In the users table, we want to access the most recent school, which is why I decided to sort before retrieving the school info.